File: app/src/main/java/org/schabi/newpipe/player/event/PlayerEventListener.java
package org.schabi.newpipe.player.event;

import com.google.android.exoplayer2.PlaybackParameters;

import org.schabi.newpipe.extractor.stream.StreamInfo;
import org.schabi.newpipe.player.playqueue.PlayQueue;

public interface PlayerEventListener {
    void onQueueUpdate(PlayQueue queue);
    void onPlaybackUpdate(int state, int repeatMode, boolean shuffled,
                          PlaybackParameters parameters);
    void onProgressUpdate(int currentProgress, int duration, int bufferPercent);
    void onMetadataUpdate(StreamInfo info, PlayQueue queue);
    default void onAudioTrackUpdate() { }
    void onServiceStopped();
}

File: app/src/main/java/org/schabi/newpipe/player/event/PlayerEventListener.kt
@file:OptIn(androidx.media3.common.util.UnstableApi::class)

package org.schabi.newpipe.player.event

import androidx.media3.common.PlaybackParameters
import org.schabi.newpipe.extractor.stream.StreamInfo
import org.schabi.newpipe.player.playqueue.PlayQueue

interface PlayerEventListener {
    fun onQueueUpdate(queue: PlayQueue?)
    fun onPlaybackUpdate(state: Int, repeatMode: Int, shuffled: Boolean, parameters: PlaybackParameters?)
    fun onProgressUpdate(currentProgress: Int, duration: Int, bufferPercent: Int)
    fun onMetadataUpdate(info: StreamInfo?, queue: PlayQueue?)
    fun onAudioTrackUpdate() {}
    fun onServiceStopped()
}
